src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,288 @@
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963");
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            // When powering the AK893, doc says 50 ms needed
+            Thread.Sleep(50);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns>The device information</returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnetometerBias { get; set; } = Vector3.Zero;
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <param name="numberMeasurements">Number of measurement for the calibration, default is 1500</param>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer(int numberMeasurements = 1500)
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magnetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            // Page 13 of documentation 
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
+            ReadBytes(Register.ASAX, rawData);
+            calib.X = (float)((rawData[0] - 128) / 256.0 + 1.0);
+            calib.Y = (float)((rawData[1] - 128) / 256.0 + 1.0);
+            calib.Z = (float)((rawData[2] - 128) / 256.0 + 1.0);
+            MeasurementMode = MeasurementMode.PowerDown;
+            MeasurementMode = oldPower;
+
+            // Now calculate the bias
+            // Store old mode to restore after
+            var oldMode = MeasurementMode;
+            Vector3 minbias = new Vector3();
+            Vector3 maxbias = new Vector3();
+
+            // Setup the 100Hz continuous mode
+            MeasurementMode = MeasurementMode.ContinuousMeasurement100Hz;
+            for (int reading = 0; reading < numberMeasurements; reading++)
+            {
+                // Timeout = 100Hz = 10 ms but during calibration, it seems that a larger value must be used
+                // TODO: find out why a higher value is needed
+                var bias = ReadMagnetometer(true, TimeSpan.FromMilliseconds(20));
+                minbias.X = Math.Min(bias.X, minbias.X);
```

Comment:
No action expected: I actually have some utilities in one of the samples to make these 6 lines collapsed into 2 (Min/Max and other helper functions for vectors) so will need to put it somewhere in common and fix all places.